modules/standard/Maps.chpl

```diff
@@ -0,0 +1,519 @@
+    }
+
+    /*
+      The current number of keys contained in this map.
+    */
+    inline proc const size {
+      _enter();
+      var result = keys.size;
+      _leave();
+      return result;
+    }
+
+    /*
+      Returns `true` if this map contains zero keys.
+
+      :returns: `true` if this map is empty.
+      :rtype: `bool`
+    */
+    inline proc const isEmpty(): bool {
+      return size == 0;
+    }
+
+    /*
+      Returns `true` if the given key is a member of this map, and `false`
+      otherwise.
+
+      :arg k: The key to test for membership.
+      :type k: keyType
+
+      :returns: Whether or not the given key is a member of this map.
+      :rtype: `bool`
+    */
+    proc const contains(const k: keyType): bool {
+      _enter();
+      var result = keys.contains(k);
+      _leave();
+      return result;
+    }
+
+    /*
+      Updates this map with the contents of the other, overwritting the values
+      for already-existing keys.
+
+      :arg m: The other map
+      :type m: map(keyType, valType)
+    */
+    proc update(const ref m: map(parSafe, keyType, valType)) {
+      _enter();
+      for key in m {
+        if !keys.contains(key) then
+          keys += key;
+        vals[key] = m.vals[key];
+      }
+      _leave();
+    }
+
+    /*
+      Get the value mapped to the given key, or add the mapping if key does not
+      exist.
+
+      :arg k: The key to access
+      :type k: keyType
+
+      :returns: Reference to the value mapped to the given key.
+    */
+    proc this(k: keyType) ref {
+      _enter();
+      if !keys.contains(k) then
+        keys += k;
+      ref result = vals[k];
+      _leave();
+      return result;
+    }
+
+    /*
+      Iterates over the keys of this map.
+
+      :yields: A reference to one of the keys contained in this map.
+    */
+    iter these() const ref {
+      for key in keys {
+        yield key;
+      }
+    }
+
+    /*
+      Iterates over the key-value pairs of this map.
+
+      :yields: A tuple of references to one of the key-value pairs contained in
+               this map.
+    */
+    iter items() const ref {
+      for key in keys {
+        yield (key, vals[key]);
+      }
+    }
+
+    /*
+      Iterates over the values of this map.
+
+      :yields: A reference to one of the values contained in this map.
+    */
+    iter values() ref {
+      for val in vals {
+        yield val;
+      }
+    }
+
+    /*
+      Writes the contents of this map to a channel. The format looks like:
+
+        .. code-block:: chapel
+    
+           {k1: v1, k2: v2, .... , kn: vn}
+
+      :arg ch: A channel to write to.
+    */
+    proc readWriteThis(ch: channel) {
+      _enter();
+      var first = true;
+      //try! {
+        ch <~> "{";
+        for key in keys {
+          if first {
+            first = false;
+          } else {
+            ch <~> ", ";
+          }
+          ch <~> key <~> ": " <~> vals[key];
+        }
+        ch <~> "}";
```

Comment:
Do we want to choose a separate bounds character here besides `{}`? @e-kayrakli What do other popular languages use to bound a map when printing it? 

It looks to me like `python` uses `{}` as well (though I already used this for `set`). Though to be honest, we're kind of out of options...

Reply:
Yes python uses `{}`. My main concern with having `[]` as set bounds was not being able to tell whether `[1,2,3]` an array or a set. But thanks to `key:value` format, I don't think `{}` is an issue.